File: java/arrays/RotateArray.java
package arrays;

public class RotateArray {

    void leftRotate(int arr[], int d, int n) {

        for (int i = 0; i < d; i++) {
            leftRotateByOne(arr, n);
        }
    }

    void leftRotateByOne(int arr[], int n) {
        int temp = arr[0];
        for (int i = 0; i < n - 1; i++) {
            arr[i] = arr[i + 1];
        }
        arr[n - 1] = temp;
    }

    void printArray(int arr[], int n) {
        for (int i = 0; i < n; i++) {
            System.out.print(arr[i] + " ");
        }
    }


    public static void main(String[] args) {
        RotateArray rotateArray = new RotateArray();
        int inputArray[] = {1, 2, 3, 4, 5, 6, 7, 8};

        rotateArray.leftRotate(inputArray, 3, 8);
        rotateArray.printArray(inputArray,8);
    }
}

File: python3/arrays/RotateArray.py
def leftRotate(arr, d, n):
    for i in range(d):
        leftRotateByOne(arr, n)


def leftRotateByOne(arr, n):
    temp = arr[0];
    for i in range(n - 1):
        arr[i] = arr[i + 1]
    arr[n - 1] = temp


def printArray(arr, size):
    for i in range(size):
        print(arr[i], end=" ")


arr = [1, 2, 3, 4, 5, 6, 7, 8]
leftRotate(arr, 4, 8)
printArray(arr, 8)
